media_driver/agnostic/common/codec/hal/codechal_encode_tracked_buffer.cpp

```diff
@@ -106,9 +106,10 @@ void CodechalEncodeTrackedBuffer::Resize()
             }
             ReleaseMvData(i);
             ReleaseDsRecon(i);
-#ifndef _FULL_OPEN_SOURCE
-            ReleaseSurfaceDS(i);
-#endif
+
+            if (m_encoder->m_openCommonKernel)
```

Comment:
sounds good to remove "common", but I think m_openKernel will be better than m_freeKernel. common means the kernel shared by all codecs

Reply:
You know, I am still confused. We will have 3 build types:
1. No kernels at all (ENABLE_KERNELS=OFF)
2. Only kernels w/ open sources are available (FREE_KERNELS=ON)
3. Full feature build when we use pre-built kernels containing all source, for some of them we will not publish sources (default build)

So, in my mind your flag m_encoder->m_openCommonKernel should be set as follows:
1. m_openCommonKernel=false for "no kernels at all (ENABLE_KERNELS=OFF)"
2. m_openCommonKernel=true for "only kernels w/ open sources are available (FREE_KERNELS=ON)"
3. m_openCommonKernel=true for "full feature build when we use pre-built kernels containing all source, for some of them we will not publish sources (default build)"

However, I am not sure that this is what you had in mind and actually implemented. Moreover, the above usage is confusing. Really, see yourself:
- In case of build 2) and 3) you should not differentiate the code which works with HME since feature presents in both open and non-open kernels
- In case of 1) you natively should expect different flag like if(m_kernelsEnabled) meaning whether any kernels are enabled at all

That's the origin of my confusion. Here is what I naturally expect to see in the code:
```
if (m_kernelsEnabled && !m_openKernels) {
  //program some kernel for which we don't publish sources
}

if (m_kernelsEnabled) {
  //program some kernel for which we publish sources: no difference for 2) and 3) since kernel is available
}

bool m_kernelsEnabled; // =true if kernels enabled in the build, =false if kernels are not enabled at all
bool m_openKernels; // =true if _only_ open kernel enabled in the build, =false if both open/closed kernels are enabled
```

Eventually you can consider to use ENABLE_KERNELS macro instead of bool variable, but idea still holds.

Reply:
generally it's what I try to implement, only difference is that for free kernel, currently we only enabled ICLLP AVC VDEnc, for other codecs and platforms, it's not enabled, that's why I am using the variable rather than the MARCO to control it. Eventually if we have the free kernels for all platforms and codecs, we definitely can replace with the MARCO FREE_KERNELS